### ISSUE TITLE
PAYPAL_INVALID_PAYMENT_OPTION when the shipping address is invalid

### ISSUE THREAD

**Issue by jortizsao** [braintree/braintree-web#411]
### General information

* SDK version: 3.40.0
* Environment: Sandbox
* Browser and OS: Chrome 71.0.3578.98 MacOS Sierra 10.13.6

### Issue description

Braintree wraps up the Paypal underneath error when the shipping address is invalid. So the error arriving to the error handler is too generic (PAYPAL_INVALID_PAYMENT_OPTION) that the only way to know the reason why Paypal rejected the checkout flow is by looking at the network tab and see the XHR response.

it would be nice to have the Paypal original error in the error handler in order to show the detailed error to the user, otherwise, the user doesn't know that the error is due to a bad zip code or city (adding an address validator before starting the Paypal flow seems not to be 100% error prune because should have the same rules as Paypal)

This is related to this https://github.com/paypal/paypal-checkout/issues/872#issuecomment-440291656

**Comment by crookedneighbor** [comment]
The original error is wrapped in the error presented:

```js
paypalCheckoutInstance.createPayment({
  flow: "checkout",
  amount: "10.00",
  currency: "USD",
  shippingAddressOverride: {
    line1: "123 foo",
    city: "flarg",
    state: "asdf"
  }
}).catch(function (err) {
  err.code // PAYPAL_INVALID_PAYMENT_OPTION

  var originalError = err.details.originalError.details.originalError.paymentResource

  originalError.errorMessage // Invalid request - see details --- country_name:Country code must be 2-character ISO 3166-1 value (upper case)|country_name:Required field missing

  originalError.errorDetails // an array of error objects
  originalError.errorDetails[0] // {field: "transactions[0].itemList.shippingAddress.countryCode", issue: "Required field missing"}
  originalError.errorDetails[1] // {field: "transactions[0].itemList.shippingAddress.countryCode", issue: "Country code must be 2-character ISO 3166-1 value (upper case)"}
})
```

(Note: in the next major version, the original error will be in the details at the top level instead of nested inside another error)

The problem seems more that PayPal is just exposing the message property of the error, and not the details that would make the error useful.

**Comment by jortizsao** [comment]
ok, I see, thanks!. So I will catch the error when creating the payment to get the details.

However, wouldn't be good to have all errors managed in the same Paypal handler? would it be possible some kind of coordination? Because from the comment in the Paypal repo, it seems that they consider this an issue in the Braintree side.

**Comment by crookedneighbor** [comment]
At this time, we're not going to make any changes on the Braintree side. (there's not really anything we can do different, other than put the original error in the message field, which means that you still miss out on the details)

It's up to checkout.js to preserve the details of the error instead of just exposing the message.
